build.gradle.kts

```diff
@@ -225,12 +227,26 @@ tasks {
 	}
 
 	javadoc {
-		// of all the javadoc checks (accessibility, html, missing, reference, syntax; see
-		// https://docs.oracle.com/javase/8/docs/technotes/tools/unix/javadoc.html#BEJEFABE)
-		// disable the warning for missing comments and tags because they spam the output
-		// (it does often not make sense to comment every tag; e.g. the @return tag on annotations)
-		(options as CoreJavadocOptions).addStringOption("Xdoclint:accessibility,html,syntax,reference", "-quiet")
-		options.encoding = "UTF-8"
+		// Exclude internal implementation package from javadoc
+		exclude("org/junitpioneer/internal")
+
+		options {
+			// Cast to standard doclet options, see https://github.com/gradle/gradle/issues/7038#issuecomment-448294937
+			this as StandardJavadocDocletOptions
+
+			links = listOf("https://junit.org/junit5/docs/current/api/")
+
+			// Set javadoc `--release` flag (affects which warnings and errors are reported)
+			addStringOption("-release", targetJavaVersion.getMajorVersion())
```

Comment:
Comment doesn't seem to match code here.

Reply:
Do you mean because of `--` vs. `-`? Gradle seems to prepend the options with a `-`, so `-release` becomes `--release`.
`javadoc` [expects `--release`](https://docs.oracle.com/en/java/javase/17/docs/specs/man/javadoc.html#options-for-javadoc) which is why I wrote that in the comment.

Reply:
Also feel free to test the changes locally. For example without the `--release` option, if one of the classes would for example contain `private int _;`, a `javadoc` error would be shown because `_` is a reserved identifier for Java 17.
However, with `--release 8` option, it only causes a warning because Java 8 still permits it.